src/System.Private.CoreLib/shared/System/Globalization/CompareInfo.cs

```diff
@@ -595,37 +595,87 @@ internal static int CompareOrdinalIgnoreCase(ref char strA, int lengthA, ref cha
         }
 
 
-        internal static bool EqualsOrdinalIgnoreCase(ref char strA, ref char strB, int length)
+        internal static bool EqualsOrdinalIgnoreCase(ref char charA, ref char charB, int length)
         {
-            ref char charA = ref strA;
-            ref char charB = ref strB;
-
-            // in InvariantMode we support all range and not only the ascii characters.
-            char maxChar = (GlobalizationMode.Invariant ? (char)0xFFFF : (char)0x7F);
-
-            while (length != 0 && charA <= maxChar && charB <= maxChar)
+            if (!GlobalizationMode.Invariant)
             {
-                // Ordinal equals or lowercase equals if the result ends up in the a-z range 
-                if (charA == charB ||
-                    ((charA | 0x20) == (charB | 0x20) &&
-                        (uint)((charA | 0x20) - 'a') <= (uint)('z' - 'a')))
+                IntPtr byteOffset = IntPtr.Zero;
+                while ((uint)length >= 2)
                 {
-                    length--;
-                    charA = ref Unsafe.Add(ref charA, 1);
-                    charB = ref Unsafe.Add(ref charB, 1);
+                    // Read 2 chars (32 bits) at a time from each string
+                    uint valueA = Unsafe.ReadUnaligned<uint>(ref Unsafe.As<char, byte>(ref Unsafe.AddByteOffset(ref charA, byteOffset)));
+                    uint valueB = Unsafe.ReadUnaligned<uint>(ref Unsafe.As<char, byte>(ref Unsafe.AddByteOffset(ref charB, byteOffset)));
+
+                    if (!Utf16Utility.AllCharsInUInt32AreAscii(valueA | valueB))
+                    {
+                        goto NonAscii; // one of the inputs contains non-ASCII data
+                    }
+
+                    // Generally, the caller has likely performed a first-pass check that the input strings
+                    // are likely equal. Consider a dictionary which computes the hash code of its key before
+                    // performing a proper deep equality check of the string contents. We want to optimize for
+                    // the case where the equality check is likely to succeed, which means that we want to avoid
+                    // branching within this loop unless we're about to exit the loop, either due to failure or
+                    // due to us running out of input data.
+
+                    if (!Utf16Utility.UInt32OrdinalIgnoreCaseAscii(valueA, valueB))
+                    {
+                        return false;
+                    }
+
+                    byteOffset += 2;
```

Comment:
This should be byteOffset += 4, which I think is causing the errors in CI. I'll also nuke the nuget package cache on my machine to see if that resolves the issue with my local tests picking up the wrong build.